junit-platform-engine/src/main/java/org/junit/platform/engine/TestEngine.java

```diff
@@ -102,6 +102,10 @@
 	 * @see #getVersion()
 	 */
 	default Optional<String> getGroupId() {
+		Optional<String> standalone = getAttribute(getClass().getClassLoader(), "Engine-GroupID-" + getId());
+		if (standalone.isPresent()) {
+			return standalone;
+		}
 		return Optional.empty();
```

Comment:
IMO this would read better with a ternary operator rather than the `if (...) { return ...; } return ...;` pattern.